src/m365/spo/commands/site/site-recyclebinitem-restore.ts

```diff
@@ -0,0 +1,79 @@
+import { Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  siteUrl: string;
+  id: string;
+}
+
+class SpoSiteRecycleBinItemRestoreCommand extends SpoCommand {
+  public get name(): string {
+    return commands.SITE_RECYCLEBINITEM_RESTORE;
+  }
+
+  public get description(): string {
+    return 'Restores given items from the site recycle bin';
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (this.verbose) {
+      logger.logToStderr(`Restoring items from recycle bin at ${args.options.siteUrl}...`);
+    }
+
+    const requestUrl: string = `${args.options.siteUrl}/_api/site/RecycleBin/RestoreByIds`;
+
+    const ids: string[] = args.options.id.split(',');
+    const idsChunks = [];
+
+    while (ids.length) {
+      idsChunks.push(ids.splice(0, 20));
+    }
+
+    idsChunks.forEach((idsChunk) => {
+      const requestOptions: any = {
+        url: requestUrl,
+        headers: {
+          'accept': 'application/json;odata=nometadata'
+        },
+        responseType: 'json',
+        data: {
+          ids: idsChunk
+        }
+      };
+
+      request
+        .post(requestOptions)
+        .then((): void => {
+          cb();
```

Comment:
We can call `cb()` just once per command as it signals to the CLI runtime that the command execution completed. In the current implementation, if we passed more than 20 IDs, the command would send completion signal but internally would keep running another iteration which is not correct. We need to update this so that we first execute all requests and only after they all succeeded call the completion callback.

Reply:
aaa yes, thanks for pointing this out 👍. This is a very good comment, didn't notice it at first also since I added the division to chunks at the very end 😉.

Reply:
done